ovs-dpdk.md

```diff
@@ -0,0 +1,170 @@
+# oVirt now supports OVS-DPDK
+
+
+[DPDK](http://dpdk.org/) (Data Plane Development Kit) provides high-performance packet processing libraries and user space drivers.
+Open vSwitch supports DPDK since v2.2 and starting with version 2.4, a DPDK-optimized vHost path is enabled in OVS.
+ 
+According to [Intel ONP performance tests](https://download.01.org/packet-processing/ONPS2.1/Intel_ONP_Release_2.1_Performance_Test_Report_Rev1.0.pdf), using OVS with DPDK can provide a huge performance enhancement, increasing network packets throughput and reduced latency.
+ 
+OVS-DPDK has been added to oVirt as an experimental feature.
+Please note that OVS-DPDK cannot access any features located within the linux kernel, for example: linux bridge,  tun/tap devices, iptables etc.
+
+## Requirements:
+
+For best performance use an Intel Xeon class server system such as Ivy Bridge, Haswell or newer.
+Ensure that each memory channel has at least one memory DIMM inserted, and that memory size should be at least 4GB.
+ 
+The network card must be on the supported-vendor matrix, located here: <http://dpdk.org/doc/nics>
+
+* Hardware support: Make sure VT-d / AMD-Vi is enabled in BIOS
+* Kerenel support: When adding a new host via oVirt GUI, go to Hosts -> Edit > Kernel and add the following parameters to Kernel Command line:  ```“iommu=pt intel_iommu=on default_hugepagesz=1GB hugepagesz=1G hugepages=N” ```
+Then צםהק the host אם maintenance mode, reboot the host and activate it when it is up again.
+
+
+
+## Configure OVS and DPDK on the Host
+
+### Install dpdk packages:
+
+```bash
+# yum install dpdk dpdk-tools driverctl
+
+```
+ 
+### Bind NICs to DPDK:
+
+In order to run a DPDK application, a suitable uio module should be loaded into kernel memory.
+Available modules are: uio_pci_generic, igb_uio and vfio-pci.
+In many cases, the standard uio_pci_generic module included in the Linux kernel can provide the uio capability.
+For some devices which lack support for legacy interrupts, igb_uio should be used, but it’s an out-of-tree kernel module.
+VFIO is preferred to uio drivers in latest DPDK versions in platforms that support VFIO.
+
+All ports that you would like to run DPDK application should be bound to one of the modules before OVS is started. DPDK will ignore all devices which are not bound to one of the drivers above.
+ 
+Load vfio-pci
+
+```
+# modprobe vfio_pci
+```
+Review the status of current port bindings:
+```
+# driverctl -v list-devices
+```
+Bind the device to DPDK:
+```
+# driverctl set-override <PCI_ID> vfio-pci
+```
+
+### Hugepages Support
+Hugepages support is required for increasing performance. By allocating hugepages, we gain larger pages, using less pages and therefore less TLB misses are involved which speeds up performance. For these reasons, 1GB hugepages will be favored.
```

Comment:
"By allocating hugepages, we gain larger pages, ..." could you maybe reword that? It's a bit obvious yet unrevealing. Also, the less TLB misses stem from different TLB for specific hugepages sizes, this whole sentence may be oversimplifying things.